carto/fields.py

```diff
@@ -87,7 +95,10 @@ def __set__(self, instance, value):
         for field in resource.fields:
             for grant_type in value:
                 if grant_type['type'] == self.type_field[field]:
-                    setattr(resource, field, grant_type[field])
+                    try:
+                        setattr(resource, field, grant_type[field])
+                    except KeyError:
+                        pass
```

Comment:
I suppose we can have errors with grants without specific fields: for example if the grant doesn't include `schemas`. Right?

I am not sure if catching the error and doing nothing could be a problem in the future. But, could we use a default one in these cases without the try/except? 
```
setattr(resource, field, grant_type.get(field, []))
```
Not sure about

Reply:
You are right... not the best solution. I had one test failing and found that as a prober solutions, but I'm going to try your suggestion since it looks like it could work 👍